### PR TITLE
2021 07 removing customization parameters

### PR DESCRIPTION
**Pull request by agenius-logesh**

### CODE REVIEW

File: R/cdfPlot.R [hunk 1]
```diff
@@ -35,63 +37,57 @@
...
-cdfPlot <- function(contrastDF,
+cdfPlot <- function(DGEdata,
+                    contrast,
                     plotType       = "canvasXpress",
                     pvalCol        = "P.Value",
                     pThreshold     = 0.01,
                     xlab,
                     ylab,
                     title          = NULL,
                     insetTitle     = NULL,
```
Comment (saranya-ag) [inline R/cdfPlot.R]: change to Viewport title

File: R/cdfPlot.R [hunk 2]
```diff
@@ -139,46 +135,12 @@ cdfPlot <- function(contrastDF,
     } else {
         if (!is.null(ylab) &&
             !all(is.character(ylab),
-                 length(ylab) == 1)) {
+            length(ylab) == 1)) {
```
Comment (saranya-ag) [inline R/cdfPlot.R]: alignment is wrong for this . revert it back to what it was.

Comment (saranya-ag) [reply R/cdfPlot.R]: Length(ylab) comes within the all function. So it should not be the same level as all

File: R/comparePlot.R [hunk 3]
```diff
@@ -67,18 +59,18 @@
...
+#'                         contrastOne,
+#'                         contrastTwo,
 #'                         pThreshold = 0.5,
 #'                         xlab = "x Axis Label",
 #'                         ylab = "y Axis Label",
-#'                         title = "Plot Title",
-#'                         crosshair = "red",
-#'                         referenceLine = "blue",
-#'                         legendPosition = "right")
+#'                         title = "Plot Title"
```
Comment (saranya-ag) [inline R/comparePlot.R]: You are missing a comma here.

**Review by saranya-ag — CHANGES_REQUESTED** [review]
@agenius-logesh  - As discussed , can you let the plot take default values for size, shape and transparency. Only in cases, where a plot is being sized by or shaped by a attribute, hardcode the values. Colors can be hardcoded.

File: R/cdfPlot.R (hunk 1 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/cdfPlot.R (hunk 2 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/comparePlot.R (hunk 3 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/cdfPlot.R [hunk 4]
```diff
@@ -26,72 +26,68 @@
 #' is required for these arguments which applies the attributes in
 #' this order: Significant, Not Significant.
 #'
-#' @param contrastDF A dataframe with LogRatio and LogIntensity columns and optionally a p-value or FDR column.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrast A character vector of a topTable data in DGEobj and its a class of dataframe
```
Comment (saranya-ag) [inline R/cdfPlot.R:30]: Wrong definition - You are passing only one contrast

File: R/cdfPlot.R [hunk 5]
```diff
@@ -26,72 +26,68 @@
...
-#' @param footnote Optional string placed right justified at bottom of plot.
 #'
 #' @return A list containing main plot, inset plot for both plotType. For plotType ="ggplot" list contains a combined plot which
 #' displays the inset plot in a viewport
 #'
 #' @examples
 #' \dontrun{
-#'    # Plot to console (contrastDF is a topTable dataframe)
-#'    cdfPlot(contrastDF, title = "My CDF Plot")
+#'    # Plot to console (DGEdata is a name of DGEonj and
```
Comment (saranya-ag) [inline R/cdfPlot.R:51]: DGEobj

File: R/cdfPlot.R [hunk 6]
```diff
@@ -26,72 +26,68 @@
...
+#'    contrast is a dataframe from DGEobj)
```
Comment (saranya-ag) [inline R/cdfPlot.R:52]: contrast is  the name of a top table from DGEobj

File: R/cdfPlot.R [hunk 7]
```diff
@@ -26,72 +26,68 @@
...
 
-    assertthat::assert_that(!missing(contrastDF),
-                            !is.null(contrastDF),
-                            "data.frame" %in% class(contrastDF),
-                            nrow(contrastDF) > 0,
-                            msg = "contrastDF must be specified as dataframe with a p-value column.")
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified as class of DGEobj.")
```
Comment (saranya-ag) [inline R/cdfPlot.R:79]: DGEdata must be specified and must belong to class DGeobj

File: R/cdfPlot.R [hunk 8]
```diff
@@ -26,72 +26,68 @@
...
+
+    assertthat::assert_that(!missing(contrast),
+                            !is.null(contrast),
+                            contrast %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            msg = "contrast to be a singular value of class character and must be one from DGEdata with LogIntensity and LogRatio columns and optionally a p-value.")
```
Comment (saranya-ag) [inline R/cdfPlot.R:84]: "contrast to be a singular value of class character and must be one of the names of the top tables in DGEdata.

File: R/comparePlot.R [hunk 9]
```diff
@@ -29,36 +29,28 @@
 #' Note: if p-values or FDR values are not used to color the plot, the X Unique color
 #' values are used.
 #'
-#' @param compareDF A dataframe with the first two columns representing the x and y variables.
-#'          Optionally add xp and yp columns to hold p-values or FDR values.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrastOne and
```
Comment (saranya-ag) [inline R/comparePlot.R:33]: There is only going to be a vector of contrast

Comment (saranya-ag) [reply R/comparePlot.R:33]: You need to update this

File: R/comparePlot.R [hunk 10]
```diff
@@ -88,31 +80,60 @@
 #' @importFrom magrittr set_rownames
 #'
 #' @export
-comparePlot <- function(compareDF,
+comparePlot <- function(DGEdata,
+                        contrastOne,
```
Comment (saranya-ag) [inline R/comparePlot.R:84]: Change this to one contrast

File: R/ggplotMDS.R [hunk 11]
```diff
@@ -448,9 +353,9 @@ ggplotMDS <- function(DGEdata,
                                               colorBy                 = colorCol,
                                               shapeBy                 = shapeCol,
                                               sizeBy                  = sizeCol,
-                                              dataPointSize           = symSize,
+                                              dataPointSize           = 10,
```
Comment (saranya-ag) [inline R/ggplotMDS.R:356]: Dont hardcode this value

File: R/ggplotMDS.R [hunk 12]
```diff
@@ -196,13 +177,6 @@ ggplotMDS <- function(DGEdata,
...
     }
 
-    if (missing(sizeBy) || is.null(sizeBy)) {
-        if (!missing(symSize) && any(is.null(symSize), length(symSize) != 1, !is.numeric(symSize))) {
-            warning("symSize must be a singular numeric value. Assigning default value 10.")
-            symSize <- 10
-        }
-    }
-
     intercept_flag   <- FALSE
```
Comment (saranya-ag) [inline R/ggplotMDS.R:180]: You dont need a intercept_flag and length here i think -if you are not customizing colors

Comment (saranya-ag) [reply R/ggplotMDS.R:180]: Can you check once more

File: R/ggplotMDS.R [hunk 13]
```diff
@@ -448,9 +353,9 @@ ggplotMDS <- function(DGEdata,
                                               colorBy                 = colorCol,
                                               shapeBy                 = shapeCol,
                                               sizeBy                  = sizeCol,
-                                              dataPointSize           = symSize,
+                                              dataPointSize           = 10,
                                               showDecorations         = TRUE,
-                                              shapes                  = shapes,
+                                              shapes                  = "circle",
```
Comment (saranya-ag) [inline R/ggplotMDS.R:358]: Dont hard code this value

File: R/obsPlot.R [hunk 14]
```diff
@@ -1,274 +1,466 @@
...
-#' @param meanColor Color for the mean layer (Default = "red2")
-#' @param meanFill Fill color for the mean layer (Default = "goldenrod1")
-#' @param meanShape Shape for the mean layer (Default = 21; fillable circle)
-#' @param meanAlpha Transparency for the mean layer (Default = 0.7)
-#' @param meanSize Size of the mean points (Default = 3)
-#' @param baseFontSize The smallest size font in the figure in points. (Default =
-#'   12)
-#' @param themeStyle "bw" or "grey" which correspond to theme_bw or theme_grey
-#'   respectively. Default = bw"
+#' @param color Color for the boxplot or the violin ploy (default = "deepskyblue3")
```
Comment (saranya-ag) [inline R/obsPlot.R:33]: Color was supposed to go away

File: R/obsPlot.R [hunk 15]
```diff
@@ -1,274 +1,466 @@
...
+                    convert_normalize   = "none",
+                    convert_prior.count = NULL,
+                    designTable         = "design",
+                    group               = "ReplicateGroup",
+                    violinLayer         = FALSE,
+                    showPoints          = TRUE,
+                    xlab,
+                    ylab,
+                    title,
+                    color               = "deepskyblue3",
```
Comment (saranya-ag) [inline R/obsPlot.R:96]: Color was supposed to go away.

File: R/obsPlot.R [hunk 16]
```diff
@@ -1,274 +1,466 @@
...
+
+    if (any(is.null(violinLayer),
+            length(violinLayer) != 1,
+            !is.logical(violinLayer))) {
+        warning("violinLayer must be a singular logical value. Assigning default value FALSE.")
+        violinLayer <- FALSE
+    }
+
+    #color Validations
+    if (any(is.null(color),
```
Comment (saranya-ag) [inline R/obsPlot.R:287]: remove color

File: R/plotDispersion.R [hunk 17]
```diff
@@ -190,33 +110,28 @@ plotDispersion <- function(DGEdata,
...
-                                                 transparency            = symbolTransparency,
+                                                 colors                  = "darkblue",
                                                  scatterOutlineThreshold = 0,
                                                  title                   = title,
                                                  yAxisTitle              = ylab,
                                                  showLoessFit            = showLoessFit,
-                                                 fitLineColor            = linefitColor,
-                                                 fitLineStyle            = lineType,
+                                                 fitLineColor            = "red",
+                                                 fitLineStyle            = "solid",
```
Comment (saranya-ag) [inline R/plotDispersion.R:119]: remove fitLineStyle

File: R/plotDispersion.R [hunk 18]
```diff
@@ -190,33 +110,28 @@ plotDispersion <- function(DGEdata,
...
+                       color = "darkblue")
 
         if (!is.null(lineFit)) {
             MyDispPlot <- MyDispPlot +
                 geom_smooth(formula  = y ~ x,
                             method   = lineFit,
-                            color    = linefitColor,
-                            linetype = lineType)
+                            color    = "red",
+                            linetype = "solid")
```
Comment (saranya-ag) [inline R/plotDispersion.R:134]: can you remove lineType

File: R/plotDispersion.R [hunk 19]
```diff
@@ -190,33 +110,28 @@ plotDispersion <- function(DGEdata,
...
 
 
     } else {
         MyDispPlot <- ggplot(plotdata, aes(x = AveLogCPM, y = Dispersion)) +
-            geom_point(size  = symbolSize,
-                       shape = symbolShape,
-                       fill  = symbolColor,
-                       color = symbolColor,
-                       alpha = symbolTransparency)
+            geom_point(shape = "circle",
```
Comment (saranya-ag) [inline R/plotDispersion.R:125]: can you remove shape

**Review by saranya-ag — CHANGES_REQUESTED** [review]

File: R/obsPlot.R (hunk 16 above)
Comment (agenius-logesh) [reply R/obsPlot.R:287]: its completed in another branch.

File: R/obsPlot.R (hunk 15 above)
Comment (agenius-logesh) [reply R/obsPlot.R:96]: its completed in another branch.

File: R/obsPlot.R (hunk 14 above)
Comment (agenius-logesh) [reply R/obsPlot.R:33]: its completed in another branch.

File: R/comparePlot.R (hunk 10 above)
Comment (agenius-logesh) [reply R/comparePlot.R:84]: its completed in another branch.

File: R/comparePlot.R (hunk 9 above)
Comment (agenius-logesh) [reply R/comparePlot.R:33]: its completed in another branch.

File: R/cdfPlot.R (hunk 8 above)
Comment (agenius-logesh) [reply R/cdfPlot.R:84]: its completed in another branch.

File: R/cdfPlot.R (hunk 7 above)
Comment (agenius-logesh) [reply R/cdfPlot.R:79]: its completed in another branch.

File: R/cdfPlot.R (hunk 6 above)
Comment (agenius-logesh) [reply R/cdfPlot.R:52]: its completed in another branch.

File: R/cdfPlot.R (hunk 5 above)
Comment (agenius-logesh) [reply R/cdfPlot.R:51]: its completed in another branch.

File: R/cdfPlot.R (hunk 4 above)
Comment (agenius-logesh) [reply R/cdfPlot.R:30]: its completed in another branch.

File: R/ggplotMDS.R (hunk 11 above)
Comment (agenius-logesh) [reply R/ggplotMDS.R:356]: if we are not hard coding this the pointsize too large

File: R/plotDispersion.R (hunk 18 above)
Comment (agenius-logesh) [reply R/plotDispersion.R:134]: Its already completed.

File: R/plotDispersion.R (hunk 17 above)
Comment (agenius-logesh) [reply R/plotDispersion.R:119]: Its already completed.

**Pull request merged**